processor/transformprocessor/internal/common/parser.go

```diff
@@ -107,7 +107,7 @@ func (n *IsNil) Capture(_ []string) error {
 	return nil
 }
 
-func ParseQueries(statements []string, functions map[string]interface{}, pathParser PathExpressionParser) ([]Query, error) {
+func ParseQueries(statements []string, functions map[string]interface{}, pathParser PathExpressionParser, enumParser EnumParser) ([]Query, error) {
```

Comment:
Should pathParser and enumParser be merged into a single interface?

Reply:
I had that originally but I wanted a way interact with Enums without having to use Getters or Setters, which PathParser returns. Since Enums are static I wanted a way to return them directly from the Contexts.